dhis-2/dhis-support/dhis-support-db-migration/src/main/java/org/hisp/dhis/db/migration/v35/V2_35_21__Generate_uid_for_sms.java

```diff
@@ -0,0 +1,107 @@
+ * modification, are permitted provided that the following conditions are met:
+ * Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ *
+ * Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ * Neither the name of the HISP project nor the names of its contributors may
+ * be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import org.flywaydb.core.api.migration.BaseJavaMigration;
+import org.flywaydb.core.api.migration.Context;
+import org.hisp.dhis.common.CodeGenerator;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+
+public class V2_35_21__Generate_uid_for_sms extends BaseJavaMigration
+{
+    private static final Logger log = LoggerFactory.getLogger( V2_35_21__Generate_uid_for_sms.class );
+
+    @Override
+    public void migrate( Context context ) throws Exception
+    {
+        List<Long> incomingSmsIds = new ArrayList<>();
+        List<Long> outboundSmsIds = new ArrayList<>();
+
+        try ( Statement statement = context.getConnection().createStatement(); ResultSet resultSetIncomingSms = statement.executeQuery( "select id from incomingsms" ) )
+        {
+            while ( resultSetIncomingSms.next() )
+            {
+                incomingSmsIds.add( resultSetIncomingSms.getLong("id") );
+            }
+        }
+
+        try ( Statement statement = context.getConnection().createStatement(); ResultSet resultSetOutboundSms = statement.executeQuery( "select id from outbound_sms" ) )
+        {
+            while ( resultSetOutboundSms.next() )
+            {
+                outboundSmsIds.add( resultSetOutboundSms.getLong( "id" ) );
+            }
+        }
+
+        if ( !incomingSmsIds.isEmpty() )
+        {
+            for ( long id : incomingSmsIds )
```

Comment:
I guess there can be a lot of SMS in a database, and this might be slow or lead to out of mem. We have a PSQL function for generating UIDs now I believe @stian-sandvold ? I would suggest we use that PSQL function instead to make this much faster and less prone to mem issues.